src/main/java/com/github/cameltooling/lsp/internal/references/ReferencesProcessor.java

```diff
@@ -99,16 +100,34 @@ private boolean isReference(CamelURIInstance camelURIInstanceToSearchReference,
 		return allCamelUriInstance;
 	}
 	
-	private String getDirectId(CamelURIInstance camelDirectURIInstance) {
+	private static String getDirectId(CamelURIInstance camelDirectURIInstance) {
 		Set<PathParamURIInstance> pathParams = camelDirectURIInstance.getComponentAndPathUriElementInstance().getPathParams();
 		if (!pathParams.isEmpty()) {
 			return pathParams.iterator().next().getValue();
 		}
 		return null;
 	}
 
-	private boolean isDirectComponentKind(CamelURIInstance camelURIInstanceToSearchReference) {
+	public static boolean isDirectComponentKind(CamelUriElementInstance camelURIInstanceToSearchReference) {
 		return POSSIBLE_DIRECT_REFERENCE.contains(camelURIInstanceToSearchReference.getComponentName());
 	}
 
+	public static List<String> retrieveEndpointIDsOfScheme(String scheme, ParserXMLFileHelper xmlFileHelper, TextDocumentItem docItem) throws Exception {
```

Comment:
this method is used only in CamelComponentSchemesCompletionsFuture, seems it would be a better fit there

Reply:
relocated into a util class